pxd.c

```diff
@@ -1035,23 +1035,14 @@ static void pxd_timeout(unsigned long args)
 		ctx->name, ctx->unique);
 }
 
-int pxd_context_init(struct pxd_context *ctx, int i)
+static void pxd_context_init(struct pxd_context *ctx, int i)
 {
-	int err;
 	spin_lock_init(&ctx->lock);
 	ctx->id = i;
 	ctx->fops = fuse_dev_operations;
 	ctx->fops.owner = THIS_MODULE;
 	ctx->fops.open = pxd_control_open;
 	ctx->fops.release = pxd_control_release;
-
-	if (ctx->id < pxd_num_contexts_exported) {
-		err = fuse_conn_init(&ctx->fc);
-		if (err)
-			return err;
-	} else {
-		ctx->fops.unlocked_ioctl = pxd_control_ioctl;
-	}
 	ctx->fc.release = pxd_fuse_conn_release;
 	ctx->fc.allow_disconnected = 1;
```

Comment:
These two need to move to pxd_context_setup(). fuse_conn_init() is going to zero out ctx->fc.